Add endpoint to fetch all users

c-bartell opened the pull request:
Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: George Soderholm 
<62966396+GeorgieGirl24@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>

### Description
Added a RESTful `GET /users` route to fetch and return data for all users. Tests and documentation still need to be added.
### Closes issue(s)
Closes #30 
### Testing
Still needs testing.
### Screenshots
![Noice screenshot of endpoint and response in postman](https://user-images.githubusercontent.com/60277914/109234913-dacc9e80-7789-11eb-9be5-daa1044453b9.png)
### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [ ] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme
- [X] I ran full test suite - all tests passing

### Other comments

c-bartell committed:
Add endpoint to fetch all users

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: George Soderholm 
<62966396+GeorgieGirl24@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>

carson-jardine commented on api/resources/users.py:
```diff
@@ -55,3 +78,13 @@ def post(self, *args, **kwargs):
+            return abort(404)
+
+        users_payload = _users_payload(users)
+        users_payload['sucess'] = True
```
this should be `success`, not `sucess`

GeorgieGirl24 committed:
Edit spelling of success in users.py

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>
